OmniKitUI/ViewControllers/ReplacePodViewController.swift

```diff
@@ -15,37 +15,50 @@ class ReplacePodViewController: SetupTableViewController {
+        case fault(_ eventType: FaultEventCode.FaultEventType)
         case canceledPairingBeforeApplication
         case canceledPairing
     }
 
     var replacementReason: PodReplacementReason = .normal {
         didSet {
-            if oldValue != replacementReason {
-                switch replacementReason {
-                case .normal:
+            switch replacementReason {
+            case .normal:
                     break // Text set in interface builder
-                case .fault:
-                    instructionsLabel.text = LocalizedString("The pod has detected an internal fault. Insulin delivery has stopped. Please remove pod and then deactivate it.", comment: "Instructions when replacing pod due to a fault")
-                case .canceledPairingBeforeApplication:
-                    instructionsLabel.text = LocalizedString("Incompletely setup pod must be deactivated before pairing with a new one. Deactivate and discard pod.", comment: "Instructions when deactivating pod that has been paired, but not attached.")
-                case .canceledPairing:
-                    instructionsLabel.text = LocalizedString("Incompletely setup pod must be deactivated before pairing with a new one.  Please remove the pod and then deactivate it.", comment: "Instructions when deactivating pod that has been paired and possibly attached.")
-                    break
+            case .fault(let eventType):
+                switch eventType {
+                case .reservoirEmpty:              // $18
+                    instructionsLabel.text = LocalizedString("Empty reservoir. Insulin delivery has stopped. Please deactivate and remove pod.", comment: "Instructions when replacing pod due to an empty reservoir")
+                case .exceededMaximumPodLife80Hrs: // $1C
+                    instructionsLabel.text = LocalizedString("Pod expired. Insulin delivery has stopped. Please deactivate and remove pod.", comment: "Instructions when replacing pod due to an expired pod")
+                case .occluded,                    // $14
+                     .occlusionCheckValueTooHigh, .occlusionCheckStartup1, .occlusionCheckStartup2,         // $5A, $60, $60
+                     .occlusionCheckTimeouts1, .occlusionCheckTimeouts2, .occlusionCheckTimeouts3,          // $62, $66, $67
+                     .occlusionCheckPulseIssue, .occlusionCheckBolusProblem, .occlusionCheckAboveThreshold: // $68, $69, $6A
+                    instructionsLabel.text = LocalizedString("Occlusion detected. Insulin delivery has stopped. Please deactivate and remove pod.", comment: "Instructions when replacing pod due to an occlusion")
+                default:
+                    instructionsLabel.text = String(format: LocalizedString("The pod has detected internal fault 0x%02x. Insulin delivery has stopped. Please deactivate and remove pod.", comment: "Instructions when replacing pod due to a fault (1: The fault code value)"), eventType.rawValue)
```

Comment:
I wonder if we should show the decimal fault code here, as that's what the pdm does.  Even though we have been tending to use the hex code in our own discussions a lot.

Reply:
I wondered about this as well, but decided to stick with using a hex numbers after talking to Katie DiSimone who suggesting doing this as we'd only be documenting to call Insulet for something like an obvious pod failure at startup and not for any screamers and so it would be better to keep our fault info more separated. The PDM errors report #'s have a lot more information and have a different format (XX-XXXXX-XXXXX-XXX) with the last 3 digits being the fault # is decimal.